mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -105,23 +107,17 @@ NetworkConfigurationFilter::resolveProxy(Http::RequestHeaderMap& request_headers
   Network::ProxyResolutionResult proxy_resolution_result = proxy_resolver->resolver->resolveProxy(
       target_url, proxy_settings_,
       [&weak_self](const std::vector<Network::ProxySettings>& proxies) {
+        ASSERT(Thread::MainThread::isMainOrTestThread());
         // This is the callback invoked from the Apple APIs resolving the PAC file URL, which
-        // happens on a separate Apple run loop thread. We keep a weak_ptr to this filter instance
+        // gets invoked on the Envoy thread. We keep a weak_ptr to this filter instance
         // so that, if the stream is canceled and the filter chain is torn down in the meantime,
         // we will fail to aquire the weak_ptr lock and won't execute any callbacks on the resolved
         // proxies.
-        if (auto caller_ptr = weak_self.lock()) {
-          Network::ProxySettingsConstSharedPtr proxy_settings =
-              Network::ProxySettings::create(proxies);
-          // We are currently in the main thread, so post the proxy resolution callback to the
-          // worker thread's (i.e. the Engine thread) dispatcher.
-          caller_ptr->decoder_callbacks_->dispatcher().post([&weak_self, proxy_settings]() {
-            // Again, the stream may have been canceled in the interim, so try aquiring the
-            // filter through its weak_ptr before proceding with the proxy resolution callback.
-            if (auto filter_ptr = weak_self.lock()) {
-              filter_ptr->onProxyResolutionComplete(proxy_settings);
-            }
-          });
+        if (auto filter_ptr = weak_self.lock()) {
+          // This call does not need to be posted on the dispatcher, because all work happens on
+          // the same thread (Envoy Mobile has one thread) and onProxyResolutionComplete posts
+          // the continueDecoding call to the dispatcher already.
```

Comment:
Your call, but since we've already verified we're on the correct thread I'm not sure this comment is necessary any more.

Reply:
done in https://github.com/envoyproxy/envoy/pull/35155